### PR TITLE
removed all references to 'multimatics' and instead discover a Generex CS121 UPS card …

### PR DESCRIPTION
**Pull request by barryodonovan**
#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [x] Have you signed the [Contributors agreement](http://docs.librenms.org/General/Contributing/)
- [x] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

…Generex card in all cases

### CODE REVIEW

**Comment by murrant** [comment]
I would prefer dropping the -card part of the os name, and maybe even the -ups part.
The icon should be more generic too, probably just Generex or even the blue diamond I see on their website.

**Comment by laf** [comment]
I'd say drop the card but leave -ups, Generex do a few other products so just in case!

The two tests that were dropped, if they will now be detected as generex then I'd say leave them in but put them in the correct test unit.

**Comment by FTBZ** [comment]
Thanks! 👍 Please note that on PR #5062 laf asked me to add some graphs in definitions.inc.php and to use snmp_get_multi() when polling.

File: tests/snmpsim/generex-ups-card.snmprec [hunk 1]
```diff
@@ -0,0 +1 @@
+1.3.6.1.2.1.1.1.0|4|CS121
```
Comment (barryodonovan) [inline tests/snmpsim/generex-ups-card.snmprec]: sysDescr on mine is STRING: "CS121 v 5.14.50"
sysObjectId on mine is OID: RFC1213-MIB::mib-2.33
@FTBZ can you confirm your sysDescr and sysObjectId. I have a feeling they will vary by firmware version but hopefully nothing more.

Comment (laf) [inline tests/snmpsim/generex-ups-card.snmprec]: That's still fine, we just should have full data to test against where possible.

**Comment by barryodonovan** [comment]
I'll redo the logo using the favicon. I'll also use snmp_get_multi in the polling.
I can add a graph or two but none of them work well with my card, It craps out due to too many snmp queries during discovery of RFC1628 sensors. I am trying to rewrite RFC1628cs121 as a module, I'll just have to figure out where to fit it in. It won't be part of this PR though.

**Comment by barryodonovan** [comment]
@laf The two dropped tests are device specific so they won't pass.

**Comment by laf** [comment]
@barryodonovan What OS will they get detected as? Whatever it is the test units just need updating to reflect them. No point in binning off valid test data.

**Comment by barryodonovan** [comment]
@laf They will get detected as Generex-ups by virtue of sysDescr and sysObjectId which are (in next push) what I do OS detection against
I am not using 1.3.6.1.2.1.33.1.1.1.0 in OS detection so having it in test data seems superfluous.

**Comment by laf** [comment]
Your focusing too much on your requirements here and not generally.

What happens when the next person adds a new OS and this uses .1.3.6.1.2.1.3.5.3.1? It will cause issues. Yes that shouldn't be possible as I expect .1.3.6.1.2.1.3 is assigned to Generex but it does happen. Test data is there to cover both existing and new OS' so having as much information as possible is better.

**Comment by barryodonovan** [comment]
@laf Sorry, don't understand your point at all.

1.3.6.1.2.1.33 is RFC1628, aka UPS-MIB.

If any device has CS121 in sysDescr and 1.3.6.1.2.1.33 in sysObjectId then it is a generex ups snmp adaptor which all use RFC1628.

If it is a motherboard model CS121 it won't be using the UPS-MIB
If it is a generic UPS using UPS-MIB, it probably won't be called CS121, but even if it is, it will have a generex logo in LibreNMS rather than a generic logo, I don't really see the problem...

**Comment by laf** [comment]
It doesn't matter what MIB an OID is part of, vendors use and abuse MIBs and OIDs - you can NOT guarantee that another vendor won't misuse 1.3.6.1.2.1.33.

The test data is to protect OS detection, having more of it helps so please just include it.

**Comment by barryodonovan** [comment]
@laf But include what?

1.3.6.1.2.1.33.1.1.1.0 is upsIdentManufacturer, a field customised by Generex for each of 99 different OEMs. We only know 4 of the possible entries and this OS doesn't even detect against it.

I am testing for two things, sysDescr and sysObjectId. People can abuse anything else they want to. If I leave it as sysDescr only, then some future motherboard will be discovered. If I leave it as sysObjectId then gamatronix, deltaUPS, etc (there are 6 OS using RFC1628 in LibreNMS currently) will all be discovered.

I can't help what came before, and I can't see what might come in the future but I also can't see what benefit there is in either including test data which is not tested nor duplicating the same test data under three different names.

**Comment by laf** [comment]
rename tests/snmpsim/multimatic.snmprec to tests/snmpsim/generex-ups1.snmprec and update the test unit, same for the other one you deleted.

**Comment by murrant** [comment]
newdevice means newdevice or updated device support ;)

File: includes/polling/os/generex-ups.inc.php [hunk 2]
```diff
@@ -0,0 +1,6 @@
+<?php
+
+$hardware_array = snmp_get_multi($device, 'upsIdentManufacturer.0 upsIdentModel.0', '-OQUs', 'UPS-MIB');
+d_echo($hardware_array);
+$hardware = trim($hardware_array[0]['upsIdentManufacturer'], '"') . ' - ' . trim($hardware_array[0]['upsIdentModel'], '"');
+$version  = snmp_get($device, 'upsIdentAgentSoftwareVersion.0', '-OQv', 'UPS-MIB');
```
Comment (Rosiak) [inline includes/polling/os/generex-ups.inc.php]: Move this to the multi get above, or are specific parameters used in the get required?

**Comment by Rosiak** [comment]
Just a single question, other than that I'm good to go!

**Comment by laf** [comment]
@barryodonovan If you can do the update @Rosiak has said about that would be good. I've put a notification out to users to say this will be merged on the 21st of December.

**Comment by laf** [comment]
Again, @librenms/reviewers please speak now if you have an issue with this.

**Comment by barryodonovan** [comment]
Change to use a single multi-get as per @Rosiak suggestion is done.
Thanks, hadn't thought of that.

**Pull request merged**